### PR TITLE
#6 Task. Punko Pavel

### PR DESCRIPTION
**Pull request by PashaPunko**
https://travis-ci.org/github/PashaPunko/js-assignments/builds/706453813

### CODE REVIEW

**Comment by PashaPunko** [comment]
https://travis-ci.org/github/PashaPunko/js-assignments/builds/706453813

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -30,7 +30,11 @@
  *
  */
 function getFizzBuzz(num) {
-    throw new Error('Not implemented');
+    var mod5 = !(num%5);
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: always prefer more code with better readablity not thecode with the least amount of charaters.

E.g. this is not much longer but more readable
let mod5 = num % 5 === 0

Do not use obsolte  **var** keyword. use either let or const

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -192,7 +217,7 @@ function findFirstSingleChar(str) {
  *
  */
 function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
-    throw new Error('Not implemented');
+    return `${isStartIncluded ? '[' : '('}${Math.min(a,b)}, ${Math.max(a,b)}${isEndIncluded ? ']' : ')'}`
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: try to make it more readable.

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -209,7 +234,12 @@ function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
  * 'noon' => 'noon'
  */
 function reverseString(str) {
-    throw new Error('Not implemented');
+    let arr = Array.from(str);
+    // arr.reverse().join('') hmmm... too easy :)
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: i'd definitely prefer arr.reverse().join('')

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -251,7 +287,16 @@ function reverseInteger(num) {
...
  */
 function isCreditCardNumber(ccn) {
-    throw new Error('Not implemented');
+    let arr = `${ccn}`.split('').reverse();
+    let sum = 0;
+    for(let i=0; i<arr.length; i++){
+        if(i%2 == 0) {
+            sum += +arr[i]; 
+        }else {
+            +arr[i]*2 > 9 ? sum += +arr[i]*2 -9 : sum += +arr[i]*2;
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: use 
sum += condition ? value1 : value2
over
condition ? sum += value1 : sum += value2

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -296,7 +349,21 @@ function getDigitalRoot(num) {
...
 function isBracketsBalanced(str) {
-    throw new Error('Not implemented');
+    let arr = Array(0);
+    
+    for (let el of str.split('')){
+        if (el == '(' || el == '[' || el == '{' || el == '<') {
+            arr.push(el);
+            continue;
+        }
+        if ((el == ')' && arr[arr.length - 1] == '(') || (el == '}' && arr[arr.length - 1] == '{') ||
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: This is too cumbersome. 

Also if you'd like to add new type of bracket you'd have make changes in 2 places. Tyr to refactor this  so you 'd have to make change only in 1 place to add a new bracket

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -332,7 +399,56 @@ function isBracketsBalanced(str) {
...
+        {begin:60*45*1000, end:60*90*1000}, 
+        {begin:60*90*1000, end:3600*22*1000}, 
+        {begin:3600*22*1000, end:3600*36*1000}, 
+        {begin:3600*36*1000, end:3600*24*25*1000}, 
+        {begin:3600*24*25*1000, end:3600*24*45*1000}, 
+        {begin:3600*24*45*1000, end:3600*24*345*1000}, 
+        {begin:3600*24*345*1000, end:3600*24*545*1000}, 
+        {begin:3600*24*545*1000, end:Number.POSITIVE_INFINITY}
+    ]
+    let f = new Map();
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: why would you want to use Map over an array here?

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -356,7 +472,12 @@ function timespanToHumanString(startDate, endDate) {
  *    365, 10 => '365'
  */
 function toNaryString(num, n) {
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js:477]: there is native funcitonlity to do this in javascript. Try to google it

**Review by AlexMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/06-conditions-n-loops-tasks.js [hunk 8]
```diff
@@ -332,7 +399,56 @@ function isBracketsBalanced(str) {
...
+        {begin:60*45*1000, end:60*90*1000}, 
+        {begin:60*90*1000, end:3600*22*1000}, 
+        {begin:3600*22*1000, end:3600*36*1000}, 
+        {begin:3600*36*1000, end:3600*24*25*1000}, 
+        {begin:3600*24*25*1000, end:3600*24*45*1000}, 
+        {begin:3600*24*45*1000, end:3600*24*345*1000}, 
+        {begin:3600*24*345*1000, end:3600*24*545*1000}, 
+        {begin:3600*24*545*1000, end:Number.POSITIVE_INFINITY}
+    ]
+    let f = new Array()
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: use array initializer

**Review by AlexMokin — COMMENTED** [review]
use spaces around operators,
do not use long lines of code


Also review this  article about code style.

https://javascript.info/coding-style

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by AlexMokin** [comment]
accepted

**Pull request closed without merging**